v2/pkg/operator/local_stored_collector.go

```diff
@@ -59,24 +63,41 @@ func (o *LocalStorageCollector) OperatorImageCollector(ctx context.Context) ([]v
 		imageIndexDir := strings.Replace(hld[len(hld)-1], ":", "/", -1)
 		cacheDir := strings.Join([]string{o.Opts.Global.WorkingDir, operatorImageExtractDir, imageIndexDir}, "/")
 		dir = strings.Join([]string{o.Opts.Global.WorkingDir, operatorImageDir, imageIndexDir}, "/")
-		if _, err := os.Stat(cacheDir); errors.Is(err, os.ErrNotExist) {
-			err := os.MkdirAll(dir, 0755)
-			if err != nil {
-				return []v1alpha3.CopyImageSchema{}, err
-			}
-			src := dockerProtocol + op.Catalog
-			dest := ociProtocolTrimmed + dir
-			err = o.Mirror.Run(ctx, src, dest, "copy", &o.Opts, *writer)
-			writer.Flush()
-			if err != nil {
-				o.Log.Error(errMsg, err)
+
+		// CLID-27 ensure we pick up oci:// (on disk) catalogs
+		if strings.Contains(op.Catalog, ociProtocol) {
+			// truncate the prefix to get the directory
+			ociDir := strings.Split(op.Catalog, ociProtocol)
+			if len(ociDir) > 0 {
```

Comment:
Since the code is checking the index [1] the len should be > 1 instead of zero. But I believe it is not needed this conditional in case image.ParseRef is used. Probably only a check if the field is not empty.